### ISSUE TITLE
BGzipCodec not found on EMR.

### ISSUE THREAD

**Issue by digitalrager** [awslabs/service-workbench-on-aws#495]
**Describe the bug**
When trying to read files using an EMR workspace I receive the following error:

Py4JJavaError: An error occurred while calling o85.json.
: java.lang.IllegalArgumentException: Compression codec is.hail.io.compress.BGzipCodec not found.

**To Reproduce**

Example code:

from pyspark.sql import SparkSession
from pyspark.sql import SQLContext

spark = SparkSession\
    .builder\
    .config("spark.sql.files.maxPartitionBytes", "1099511627776")\
    .config("spark.hadoop.mapreduce.input.fileinputformat.split.minsize", "1099511627776")\
    .config("spark.hadoop.parquet.block.size", "1099511627776")\
    .getOrCreate()

df = spark.read.json('example-files-json/*.json')

If the codecs are specified and do not include BGzipCodec the error is avoided:
 
.config("spark.hadoop.io.compression.codecs", "org.apache.hadoop.io.compress.DefaultCodec,org.apache.hadoop.io.compress.GzipCodec")\

**Expected behavior**

The dataframe should load from hdfs. 

**Versions (please complete the following information):**
We are on version 2.1.5
EMR: emr-5.27.0
 Hive 2.3.5, Spark 2.4.4

**Comment by Bingjiling** [comment]
Hi @digitalrager , thanks for reaching out! Just so I understand the question correctly: 

This works: 
``` 
.config("spark.hadoop.io.compression.codecs", "org.apache.hadoop.io.compress.DefaultCodec,org.apache.hadoop.io.compress.GzipCodec")
```

And this throws error: 
```
.config("spark.hadoop.io.compression.codecs", "org.apache.hadoop.io.compress.DefaultCodec,is.hail.io.compress.BGzipCodec,org.apache.hadoop.io.compress.GzipCodec")
```

Is that correct?

**Comment by digitalrager** [comment]
Thanks for the reply.  When I use the config which excludes the codec it works.  When I do not apply any config with regards to the codecs it fails.  I have not tried specifying BGzipCodec as I was simply trying to exclude it.

**Comment by jn1119** [comment]
Thanks @digitalrager it does seem like BGzipCodec is not installed properly. Action item on our side would be to correctly install Hail which doesn't seem to be the case. Meanwhile, I will speak with my team and see if we can remove `BGzipCodec` from defaults since it doesn't work anyways.

**Comment by jn1119** [comment]
Fix has been released in version [3.2.0](https://github.com/awslabs/service-workbench-on-aws/releases/tag/v3.2.0)
